stabilizer/src/main/java/com/hazelcast/stabilizer/worker/selector/OperationSelectorBuilder.java

```diff
@@ -0,0 +1,97 @@
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Builder to create an {@link OperationSelector} instance. Each operations is represented as an
+ * enum item and is registered with its probability (0.0 to 1.0) via the {@link #addOperation(Enum, double)} method.
+ * <p/>
+ *
+ * @param <T> enum of operations
+ */
+public class OperationSelectorBuilder<T extends Enum<T>> {
+    private final Map<T, Double> operations;
+    private T defaultOperation;
+    private double remaining;
+
+    public OperationSelectorBuilder() {
+        operations = new HashMap<T, Double>();
+        remaining = 1;
+    }
+
+    public OperationSelectorBuilder(OperationSelector<T> selector) {
+        this();
+        for (Map.Entry<T, Double> entry : selector.getOperations().entrySet()) {
+            addOperation(entry.getKey(), entry.getValue());
+        }
+        defaultOperation = selector.getDefaultOperation();
+    }
+
+    /**
+     * Register new operation for selection.
+     *
+     * @param operation   operation to be selected
+     * @param probability probability of this operation. It must be a non-negative number between 0.0 and 1.0.
+     * @return this instance to allow method-chaining
+     */
+    public OperationSelectorBuilder<T> addOperation(T operation, double probability) {
+        if (operations.put(operation, probability) != null) {
+            throw new IllegalStateException("Operations " + operation + " has been already added to this selector.");
+        }
+        remaining -= probability;
+        if (remaining < -0.1) {
+            return onProbabilityExceeded();
+        }
+        return this;
+    }
+
+    /**
+     * Register new operation for selection which uses the remaining probability.
+     *
+     * @param operation operation to be selected
+     * @return this instance to allow method-chaining
+     */
+    public OperationSelectorBuilder<T> addOperationRemainingProbability(T operation) {
+        if (operations.put(operation, remaining) != null) {
+            throw new IllegalStateException("Operations " + operation + " has been already added to this selector.");
+        }
+        remaining = -1;
+        return this;
+    }
+
+    /**
+     * Register an operation to be returned when no operation registered via {@link #addOperation(Enum, double)} or
+     * {@link #addOperationRemainingProbability(Enum)} has been selected.
+     *
+     * @param operation operation to be selected if no other operation is configured
+     * @return this instance to allow method-chaining
+     */
+    public OperationSelectorBuilder<T> addDefaultOperation(T operation) {
+        this.defaultOperation = operation;
+        return this;
+    }
+
+    /**
+     * Returns a thread-safe instance of {@link com.hazelcast.stabilizer.worker.selector.OperationSelector}.
+     *
+     * @return instance of OperationSelector
+     */
+    public OperationSelector<T> getOperationSelector() {
```

Comment:
what about something like build() ? 
I believe that's a common name for a builder pattern.